core/domain/recommendations_validators.py

```diff
@@ -0,0 +1,111 @@
+# coding: utf-8
+#
+# Copyright 2020 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Validators for recommendation services in prod."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from core.domain import base_model_validators
+from core.domain import recommendations_services
+from core.platform import models
+import python_utils
+
+(
+    exp_models, recommendations_models) = (
+        models.Registry.import_models([
+            models.NAMES.exploration,
+            models.NAMES.recommendations]))
+
+
+class ExplorationRecommendationsModelValidator(
+        base_model_validators.BaseModelValidator):
+    """Class for validating ExplorationRecommendationsModel."""
+
+    @classmethod
+    def _get_external_id_relationships(cls, item):
+        return [
+            base_model_validators.ExternalModelFetcherDetails(
+                'exploration_ids', exp_models.ExplorationModel,
+                [item.id] + item.recommended_exploration_ids)]
+
+    @classmethod
+    def _validate_item_id_not_in_recommended_exploration_ids(cls, item):
+        """Validate that model id is not present in recommended
+        exploration ids.
+
+        Args:
+            item: datastore_services.Model. ExplorationRecommendationsModel
```

Comment:
Is this an `ExplorationRecommendationsModel` instance, if so have that as type here.

Reply:
hi - I am not sure what you mean .. do you mean adding typing to the function. Cause I thought item: datastore_services.Model. ExplorationRecommendationsModel in args gives more info. Could you elaborate please

Reply:
Seems like all of the validator classmethods in `core/domain/prod_validators.py` follow this format for the Args:
```
item: datastore_services.Model. xModel to validate.
```
Maybe we should keep this block of code the way it is unless we want to do a uniform change across all of the validator classmethods in a later PR? @aks681

Reply:
Oh, the whole thing represents the model? It's fine then. Because of the space, I thought the second part was the description.
Can you remove the space before ExplorationRecommendationModel then?